iotilesensorgraph/iotile/sg/graph.py

```diff
@@ -317,6 +319,101 @@ def get_tick(self, name):
             return 0
 
+    def _parse_configs_ignorelist(self):
+        """Parses the string containing configs to ignore into a list of tuples
+
+        Returns:
+            list(tuple): A list of config variable entries that are formatted
+                as tuples (slot, config_id)
+        """
+
+        configs_ignorelist = []
+        slot = SlotIdentifier.FromString('controller')
+
+        configs_ignorelist_string = self.metadata_database['hash_configs_ignorelist']
+        configs_ignorelist_address = self.metadata_database['hash_configs_ignorelist_address']
+        configs_ignorelist_type, _ = self.get_config(slot, configs_ignorelist_address)
+        self.add_config(slot, configs_ignorelist_address, configs_ignorelist_type, configs_ignorelist_string)
+
+        ignore_entries = configs_ignorelist_string.split(',')
+        for entry in ignore_entries:
+            target, entry = entry.split(':')
+            slot = SlotIdentifier.FromString(target)
+            configs_ignorelist.append((slot, int(entry, 0)))
+
+        return configs_ignorelist
+
+    def add_checksum(self):
+        """Check metadata if sensorgraph's checksum needs to be added.
+
+        The sensorgraph must contain all the meta tags required to calculate the
+        checksum.
+
+        hash_address: Address where calculated checksum will be flashed.
+        hash_algorihm: Algorithm to use to calculate checksums, only "sha256"
+            and "crc32_0x104C11DB7" are currently supported.
+        hash_algorithm_address: Address where the algorithm specifier is
+            flashed. This is used so the programmer knows which algorithm to use
+            to debug the device.
+        hash_configs_ignorelist: List of config variables to exclude during the
+            calculation of checksum. This MUST contain the 'hash_address'.
+            The ignore list must be formatted like so...
+                meta hash_configs_ignorelist = "controller:0xcafe,slot 1:0xbabe";
+            Where each ignored variable is <target>:<config_id>, delimitted by a
+            comma in between each entry.
+        hash_configs_ignorelist_address: Address where the ignore list is
+            flashed. This is used so the programmer knows which config variables
+            to exclude in the calculation of the device's checksums.
+        """
+
+        if 'hash_address' not in self.metadata_database or\
+            'hash_algorithm' not in self.metadata_database or\
+            'hash_algorithm_address' not in self.metadata_database or\
+            'hash_configs_ignorelist' not in self.metadata_database or\
+            'hash_configs_ignorelist_address' not in self.metadata_database:
+            return
+
+        slot = SlotIdentifier.FromString('controller')
+        hash_address = self.metadata_database['hash_address']
+        algorithm = self.metadata_database['hash_algorithm']
+        algorithm_address = self.metadata_database['hash_algorithm_address']
```

Comment:
That is what I had done initially. However, @ekkizogloy pointed out that the implementation is not too obvious, and it can be limiting. I see validity in both perspectives, and I feel like its just preference at this point.